x-pack/plugins/monitoring/server/lib/elasticsearch/verify_monitoring_auth.ts

```diff
@@ -17,6 +17,8 @@ import { LegacyRequest } from '../../types';
  *
  * @param req {Object} the server route handler request object
  */
+
+// TODO: replace LegacyRequest with current request object + plugin retrieval
```

Comment:
Might be good to open an issue for this too. I think it's related to https://github.com/elastic/kibana/issues/109232 but I can't find a stack-monitoring-specific issue.